git_machete/git_operations.py

```diff
@@ -911,6 +958,22 @@ def get_commits_between(self, earliest_exclusive: AnyRevision, latest_inclusive:
             utils.get_non_empty_lines(self._popen_git("log", "--format=%H:%h:%s", f"^{earliest_exclusive}", latest_inclusive, "--").stdout)
         ))))
 
+    def get_diffs_between(self, earliest_exclusive: AnyRevision, latest_inclusive: AnyRevision) -> Dict[FullCommitHash, str]:
+        # Returns a dictionary of git hashes and their diffs between two revisions
+        marker = "__MACHETE___COMMIT_MARKER__"  # Use a marker so we can split the output into individual diffs for each commit
```

Comment:
Hmmm is it possible to just pipe the entire output of `git log --patch` to `git patch-id`? and then you should get `(patch SHA, commit SHA)` pairs 🤔 no parsing of the output of `git log --patch` needed

Reply:
If possible, this should speed up the entire thing a little bit more (less `git` process invocations)

Reply:
`git patch-id` seems to be very low-level, I didn't find any way to make it output one patch ID per block of input. We could do something with AWK and some bash magic, the number of `git` process invocations would be the same although they wouldn't be done from Python.

Reply:
Hmmm could you check `git log -p | git patch-id` on `master` in this repo?

I'm getting something like
```
3fdf24106a4e04a993e33ab4cf267980e90d1f48 cbd9cf96fe298e3339fd00553d758c5e48347c62
8134efef8d2173a007a246b4afcd952839bd338a 2a8ed18196145b15f2e2816f66f9e76ea77e5f7f
e4f8fe513423413e89920af61c13715b463bf703 ace2593d9d25144d8153acc33700228bcca2ff27
1a2813c326d41d10a041742dbf367ce0fa47540a 33808252399b358d068c63651bb23c9b3ab465c4
0df316d8afe3d3f9b2acd9f8fe33f9fedfb4c386 fe48b5e51299cbfc143b7b3051f6eeb84b4c95f4
6487ae2cc44a597e9122f6e146d90f3654476f46 fba21dfb749f5af7ccd8f76fb41a72ce91903312
75e74c3587ec1cdcc260c538bf5d94f6ccaa57ad 40531ec669fa535c2993a52dbfe8d6db9850b29d
...
```

which seems to be `(patch hash, commit hash)` pairs... would it be applicable for this use? 🤔

Reply:
It definitely is... I didn't think this was possible from reading the docs, I didn't think of actually trying :D

Reply:
Okay! lemme now try it on my large commercial repo. I'll do the final touches (any documentation etc. fixes needed)

Reply:
☝🏻 the repo has ~800k commits and ~400k tracked files... the execution times `git log --patch` are in the order of 2-4 seconds **each**. So indeed, let's keep it behind a flag/config key, as you implemented it (and not make a default) 😅

Reply:
Heh, yeah, it's quite a bit. I merged the latest develop onto this branch and also fixed some errors in CI, and checked that indeed simple mode is the default.